@here/harp-mapview/lib/PolarTileDataSource.ts

```diff
@@ -127,15 +124,13 @@ export class PolarTileDataSource extends DataSource {
     }
 
     /** @override */
-    setStyleSet(styleSet?: StyleSet, definitions?: Definitions, languages?: string[]): void {
+    setStyleSet(options: StyleSetOptions): void {
```

Comment:
to keep some basic "backward source compatibility" we could overload the setStyleSet signature (many of tests are setStyleSet), e.g. we could use as signature:

```
    setStyleSet(options: StyleSetOptions | StyleSet): void {
       if (Array.isArray(options)) {
           // a StyleSet was passed
           options = { styleSet: options }; // wrap it to StyleSetOptions
       }
       // use options
    }
```

as you pointed out it is a bit confusing that setStyleSet takes a StyelSetOptions as argument.

Reply:
As its deprecated anyway, not sure we should put effort into backwards compatibility, I also have a PR almost ready to actually deprecate DataSource.setStyleSet, then this will dissapear anyway

Reply:
added backwardscompatibility